src/passes/OptimizeInstructions.cpp

```diff
@@ -205,6 +215,182 @@ struct OptimizeInstructions
       if (isSymmetric(binary)) {
         canonicalize(binary);
       }
+    }
+
+    {
+      // TODO: It is an ongoing project to port more transformations to the
+      // match API. Once most of the transformations have been ported, the
+      // `using namespace Match` can be hoisted to function scope and this extra
+      // block scope can be removed.
+      using namespace Match;
+      Builder builder(*getModule());
+      {
+        // X == 0 => eqz X
+        Expression* x;
+        if (matches(curr, binary(EqInt32, any(&x), i32(0)))) {
```

Comment:
we could avoid the `&` here by using `Expression*&` in the API, which would be shorter? or is the idea this is more explicit?

Reply:
Yeah, I actually like having `&` as a sigil to differentiate binders for extracted values. It also allows us to provide more functionality, e.g. having both `i32(int32_t)`, which matches exact values, and `i32(int32_t*)` which matches and extracts any value.